src/middlewared/middlewared/plugins/system.py

```diff
@@ -23,6 +24,165 @@
 SYSTEM_READY = False
 
 
+class SytemAdvancedService(ConfigService):
+
+    class Config:
+        datastore = 'system.advanced'
+        datastore_prefix = 'adv_'
+        datastore_extend = 'system.advanced.system_advanced_extend'
+        namespace = 'system.advanced'
+
+    @accepts()
+    async def serial_port_choices(self):
+        if(
+            not await self.middleware.call('system.is_freenas') and
+            await self.middleware.call('notifier.failover_hardware') == 'ECHOSTREAM'
+        ):
+            ports = ['0x3f8', '0x3f8']
```

Comment:
Here you probably just want `ports = ['0x3f8']`

Legacy UI was repeating it because it was a tuple there, I believe.